src/Pagination.vue

```diff
@@ -57,6 +59,11 @@ export default {
     }
   },
   methods: {
+    checkPageCounts() {
+      if (this.curPage > this.totalPage && (this.curPage !== 1)) {
+        this.handleClick(this.curPage - 1)
+      }
+    },
     handleClick (n) {
       this.query.offset = (n - 1) * +this.query.limit
```

Comment:
I think the developer is mutating a field of parent component here and all the paginating stuff rely on this mutation. So, the design requires to use this handleClick method even it is a bad practice in my opinion.